nipoppy/data_api.py

```diff
@@ -60,59 +81,76 @@ def _check_derivatives_arg(
                     f", got invalid specification {derivatives_spec}"
                 )
 
+    def _get_phenotypes_table(self, phenotypes: List[str]) -> pd.DataFrame:
+        return self._load_tsv(
+            self.study.layout.fpath_harmonized, index_cols=self.index_cols_phenotypes
+        ).loc[:, phenotypes]
+
     def _get_derivatives_table(
         self,
         pipeline_name: str,
         pipeline_version: str,
         filepath_pattern: str,
     ) -> pd.DataFrame:
-        candidate_paths = list(
+        candidate_path = list(
             self.study.layout.get_dpath_pipeline(
                 pipeline_name=pipeline_name, pipeline_version=pipeline_version
             ).glob(filepath_pattern)
         )
-        if len(candidate_paths) == 0:
+        if len(candidate_path) == 0:
             raise FileNotFoundError(
                 f"No file matching {filepath_pattern} for pipeline "
                 f"{pipeline_name}, version {pipeline_version}"
             )
-        elif len(candidate_paths) > 1:
+        elif len(candidate_path) > 1:
             raise RuntimeError(
                 f"Found more than one file matching {filepath_pattern} for pipeline "
-                f"{pipeline_name}, version {pipeline_version}: {candidate_paths}"
+                f"{pipeline_name}, version {pipeline_version}: {candidate_path}"
             )
-        return self._load_tsv(candidate_paths.pop(), index_cols=self.imaging_index_cols)
+        return self._load_tsv(
+            candidate_path.pop(), index_cols=self.index_cols_derivatives
+        )
 
     def get_tabular_data(
```

Comment:
Would be great to have separate functions for `get_derivatives` and `get_phenotypes`. Those functions could be reuse here to merge both tables